### PR TITLE
Updated the netherworld creatures' sprites

### PR DESCRIPTION
**Pull request by Emanthealmighty**
**What does this PR do:**
Updates the netherworld creatures' sprites.
Added some more random sounds for the migo.

Credits to Ionward.

**Changelog:**
:cl:
tweak: Updated the netherworld creatures' sprites.
/:cl:

### CODE REVIEW

**Comment by MrMagolor** [comment]
Shouldn't there be pictures of the sprite changes?

**Comment by Pixelzilla** [comment]
I do like the updated creature sprites. I have a few questions regarding the mi-go. 

1.) Is the anti-aliasing around the head/knees/claw/arm intentional and if so what purpose does it serve? It didn't seem very noticeable unless you look at the sprite in an editor and zoom in.

2.) The movement sprite is quite animated with the exception of the arm/claw/tail (not sure which it is!). And i find myself drawn to that particular spot because it is static when the creature moves and that seems out of place. Could this be fixed? (if not, i did do a quick edit of it if you'd like that)

**Comment by Fox-McCloud** [comment]
You can't just say "updated the code" without context.

**Comment by Fox-McCloud** [comment]
@Ionward 

Any reason why some of the sprites are as @Pixelzilla said?

**Comment by Fox-McCloud** [comment]
I'm giving you a couple more days do do anything related to this PR or provide any feedback, as it's almost been a week without anything said.

**Comment by Emanthealmighty** [comment]
As a matter of fact I have forwarded what Pixelzilla said regarding the sprites to the one who actually made them, and I've been sent the updated and fixed sprites. It's just been a matter of getting to updating this.

**Comment by Emanthealmighty** [comment]
The sprites were adjusted.

**Comment by KasparoVy** [comment]
Looking at the PR it looks like the the new sprites were omitted completely

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
 	desc = "A sanity-destroying otherthing from the netherworld."
-	icon_state = "otherthing"
-	icon_living = "otherthing"
-	icon_dead = "otherthing-dead"
-	health = 80
-	maxHealth = 80
+	icon_state = "otherthing-pink"
+	icon_living = "otherthing-pink"
+	icon_dead = "otherthing-pink-dead"
+	health = 125
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Why their health has been buffed?

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
+	maxHealth = 125
 	obj_damage = 100
 	melee_damage_lower = 25
-	melee_damage_upper = 50
-	attacktext = "slashes"
+	melee_damage_upper = 30
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Why the damage nerf?

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
+	attacktext = "chomps"
 	attack_sound = 'sound/weapons/bladeslice.ogg'
-	faction = list("creature")
 	speak_emote = list("screams")
 	gold_core_spawnable = CHEM_MOB_SPAWN_HOSTILE
 	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
 	minbodytemp = 0
-	faction = list("nether")
+	faction = list("nether", "mining", "cult")
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Adding faction means they don't attack. I can understand why you add mining since necropolis tendril is a mining faction mob.

However I shouldn't have to guess *why* you're making changes.

Why did you add cult?

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 4]
```diff
@@ -1,22 +1,32 @@
...
+	vision_range = 2
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	unsuitable_atmos_damage = 15
+	weather_immunities = list("lava","ash")
+	obj_damage = 30
+	environment_smash = 2
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Why add the ability to destroy object?

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 5]
```diff
@@ -1,22 +1,32 @@
...
 	speak_emote = list("screams")
 	gold_core_spawnable = CHEM_MOB_SPAWN_HOSTILE
 	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
 	minbodytemp = 0
-	faction = list("nether")
+	faction = list("nether", "mining", "cult")
+	vision_range = 2
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	unsuitable_atmos_damage = 15
+	weather_immunities = list("lava","ash")
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Understandable but what kind of design problem you see with them having no immunities? Explain.

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 6]
```diff
@@ -1,22 +1,32 @@
...
+	obj_damage = 30
+	environment_smash = 2
+	minbodytemp = 0
+	heat_damage_per_tick = 20
+	see_in_dark = 8
+	see_invisible = SEE_INVISIBLE_MINIMUM
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Why?

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 7]
```diff
@@ -1,22 +1,32 @@
...
+	mob_size = MOB_SIZE_LARGE
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: This'll make crusher work on them, why?

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 8]
```diff
@@ -70,28 +80,43 @@
 	icon_state = "nether"
 	max_integrity = 50
 	spawn_time = 600 //1 minute
-	max_mobs = 15
+	max_mobs = 10
 	icon = 'icons/mob/nest.dmi'
 	spawn_text = "crawls through"
 	mob_types = list(/mob/living/simple_animal/hostile/netherworld/migo, /mob/living/simple_animal/hostile/netherworld, /mob/living/simple_animal/hostile/netherworld/blankbody)
-	faction = list("nether")
+	faction = list("nether", "mining", "cult")
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: You get it, why

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 9]
```diff
@@ -70,28 +80,43 @@
...
 
 /obj/structure/spawner/nether/Initialize(mapload)
 	.=..()
 	START_PROCESSING(SSprocessing, src)
 
 /obj/structure/spawner/nether/examine(mob/user)
 	..()
-	if(isskeleton(user) || iszombie(user))
-		to_chat(user, "A direct link to another dimension full of creatures very happy to see you. <span class='nicegreen'>You can see your house from here!</span>")
+	if(iscultist(user))
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Why did you remove the fluff text for skeleton and zombie, and replace it for cultists only.

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 10]
```diff
@@ -70,28 +80,43 @@
...
+		to_chat(user, "A direct link to another dimension full of creatures very happy to see your God again. <span class='notice'>They are patiently waiting to be re-enlisted.</span>") //By touching the portal on harm or disarm intent, you will recall all the monsters spawned by the portal. By touching it on help or grab intent, you will be able to move it around.</span>")
 	else
 		to_chat(user, "A direct link to another dimension full of creatures not very happy to see you. <span class='warning'>Entering the link would be a very bad idea.</span>")
 
 /obj/structure/spawner/nether/attack_hand(mob/user)
 	. = ..()
-	if(isskeleton(user) || iszombie(user))
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Again, why remove this?

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 11]
```diff
@@ -70,28 +80,43 @@
...
+				to_chat(user, "<span class='notice'>You anchor the portal in place!</span>")
+				move_resist = initial(move_resist)
+				user.stop_pulling()
+				spawn_time = initial(spawn_time)
+		if((user.a_intent == "harm") || (user.a_intent == "disarm")) // Kill all the creatures so they can be respawned where needed, but damage the portal...
+			for(var/mob/living/M in spawned_mobs)
+				if(M)
+					M.gib()
+			to_chat(user, "<span class='notice'>All the monsters spawned by the portal disappear in an explosion of gore, returning to the nether!</span>")*/
+	if(!iscultist(user))
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: This'll yank and pull skeleton and zombies

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 12]
```diff
@@ -70,28 +80,43 @@
...
+		to_chat(user, "A direct link to another dimension full of creatures very happy to see your God again. <span class='notice'>They are patiently waiting to be re-enlisted.</span>") //By touching the portal on harm or disarm intent, you will recall all the monsters spawned by the portal. By touching it on help or grab intent, you will be able to move it around.</span>")
 	else
 		to_chat(user, "A direct link to another dimension full of creatures not very happy to see you. <span class='warning'>Entering the link would be a very bad idea.</span>")
 
 /obj/structure/spawner/nether/attack_hand(mob/user)
 	. = ..()
-	if(isskeleton(user) || iszombie(user))
-		to_chat(user, "<span class='notice'>You don't feel like going home yet...</span>")
-	else
+/*	if(iscultist(user)) // Touched by cultist
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: A massive block of commented out code added here. Why?

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 13]
```diff
@@ -101,11 +126,44 @@
...
+				blank.melee_damage_upper = 15
+				visible_message("<span class='warning'>[M.real_name] reemerges from the link!</span>")
+				if((blankckey) && (blankckey == M.ckey))
+					blank.ckey = blankckey
+					blankckey = null
+					to_chat(blank, "<span class='biggerdanger'>Hatred courses through you... you can't give up your spirit... not yet. Help your newfound compatriots complete their objectives!</span>")
+					SSticker.mode.add_cultist(blank.mind, 1)
+					blank.mind.special_role = "Cultist"
+				qdel(M)
+				blankckey = null
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: So.

Why'd you decide that putting an antag giver (Which is prone to abuse and is why wish granter was made a very hard to enter ruin in the first place) a good design idea?

This encourages people to actively seek out netherworld portal and legitimately self-antag

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 14]
```diff
@@ -101,11 +126,44 @@
...
+			else
+				G.affecting.loc = src.loc
+				G.affecting.Weaken(5)
+				visible_message("<span class='warning'>[G.assailant] shoves [G.affecting] into the portal!</span>")
+				contents.Add(G.affecting)
+				blankckey = G.affecting.ckey // Doing it on purpose will let the person control the blank body
+				to_chat(G.affecting, "<span class='biggerdanger'>ARGHHH--</span>")
+		else
+			to_chat(user, "<span class='warning'>Come here yourself!</span>")
+			return*/
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Another big block of commented out code added.

You need to have really, really good reason to leave in non-functional, commented out feature. Else it is codebloat that another person will have to figure out why did you add and when to uncomment them

Comment (Emanthealmighty) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: I have added these features to see what people would think of them, though I realize now I should have been more clear than just saying "added in some experimental commented out features".

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm (hunk 11 above)
Comment (Emanthealmighty) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: I changed it to give a special message to cultists rather than zombies or skeletons. 

The idea was that cultists could do _more_ with the portal, which is what the commented out features are, however if they are completely removed later on (which they probably will) I'll revert it back to just giving a special message for skeletons and zombies.

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm (hunk 10 above)
Comment (Emanthealmighty) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: I changed it to give a special message to cultists rather than zombies or skeletons.

The idea was that cultists could do more with the portal, which is what the commented out features are, however if they are completely removed later on (which they probably will) I'll revert it back to just giving a special message for skeletons and zombies.

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm (hunk 5 above)
Comment (Emanthealmighty) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: I wanted to make them more in line with the other mining mobs.

**Comment by TDSSS** [comment]
I'm using netherworld portals in my own open PR, redactors don't matter but if you add new features like cult interaction, can you make those on a new subtype of the portal so I am not affected?

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 15]
```diff
@@ -1,22 +1,24 @@
...
+	icon_state = "otherthing-pink"
+	icon_living = "otherthing-pink"
+	icon_dead = "otherthing-pink-dead"
 	health = 80
 	maxHealth = 80
 	obj_damage = 100
 	melee_damage_lower = 25
 	melee_damage_upper = 50
-	attacktext = "slashes"
+	attacktext = "chomps"
```
Comment (Couls) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: No reason for this

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 16]
```diff
@@ -1,22 +1,24 @@
...
 	attack_sound = 'sound/weapons/bladeslice.ogg'
 	faction = list("creature")
 	speak_emote = list("screams")
 	gold_core_spawnable = CHEM_MOB_SPAWN_HOSTILE
 	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
 	minbodytemp = 0
-	faction = list("nether")
+	faction = list("nether", "mining")
```
Comment (Couls) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Why change its faction?

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 17]
```diff
@@ -1,22 +1,24 @@
...
+	see_in_dark = 8
```
Comment (Couls) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: undocumented

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 18]
```diff
@@ -1,22 +1,24 @@
...
+	see_invisible = SEE_INVISIBLE_MINIMUM
```
Comment (Couls) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: undocumented

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm (hunk 15 above)
Comment (variableundefined) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: I said this one is fine to fit the new sprite, in his defense

**Comment by variableundefined** [comment]
Rest are still undocumented

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm (hunk 15 above)
Comment (Couls) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Ah I got it, can resolve then.

**Comment by variableundefined** [comment]
Quick fix.

Please fix within four days.

**Comment by Emanthealmighty** [comment]
Sorry for all the trouble I've caused lately but... Fix _what_ exactly?

**Comment by variableundefined** [comment]
The faction, see in dark and see invisible changes.

**Comment by variableundefined** [comment]
@Emanthealmighty Icons conflicts. Otherwise ready to go
